core/templates/pages/exploration-player-page/services/command-executor.service.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Oppia interaction from outer iframe with messaging commands.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import {WindowRef} from 'services/contextual/window-ref.service.ts';
+import {WindowWrapperMessageService} from
+  'pages/exploration-player-page/services/window-wrapper-message.service';
+require('pages/exploration-player-page/services/exploration-engine.service.ts');
+import {ServicesConstants} from 'services/services.constants';
+
+@Injectable({providedIn: 'root'})
+export class CommandExecutorService {
+  private commandToFunctionMap = {
```

Comment:
Add single line comments above some of these explaining what they mean. For example, some context that this is the complete set of commands supported by the CommandExecutorService would be helpful.